tutorials/3d/portals/appendix.rst

```diff
@@ -0,0 +1,87 @@
+
+.. _doc_rooms_and_portals_blender:
+
+Creating room systems in Blender (or other modeling tools)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Although you can create your room system entirely within the editor, you can also build rooms and portals within your modeling tool. There is one small snag - modeling tools such as Blender have no knowledge of Room, RoomGroup and Portal nodes. In order to work around this we use the naming conventions that were discussed earlier.
+
+Rooms should be created as Empties with names such as ``Room_kitchen`` (substituting your preferred room name). If you place meshes within the rooms within the modeling tool, these Empties will form Spatial nodes in Godot, and will be converted to room nodes automatically during the room conversion phase.
+
+Creating a RoomGroup is easy - any ``Empty`` with the name prefix ``RoomGroup`` will be converted to a RoomGroup.
+
+Creating Portals in modeling tools
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Portals also use the same naming convention as within Godot - the name should start with the prefix ``Portal_`` followed by the name of the room the portal should link to. However, for portals, we want to specify the shape and position of the polygon within the modeling tool, so the nodes should be ``Meshes`` rather than ``Empties``.
+
+Portal meshes have some restrictions to work properly. They should be convex, and the polygon points should be in the same plane. The accuracy to the plane does not have to be exact, as Godot will automatically average the direction of the portal plane. Once converted to a :ref:`Portal<class_Portal>` node, the snapping to the portal plane is enforced, and the vertices are specified (and editable) as 2d coordinates in the inspector, rather than 3d points.
+
+Portal point editing
+~~~~~~~~~~~~~~~~~~~~
+Portals are defined by a combination of the transform of the portal node, and by a set of points which form the corners.
+
+The default portal has 4 corners as shown in the inspector:
+
+.. image:: img/portal_points.png
+
+You can edit these points in the inspector to make a better match to the opening in your game level. But bear in mind it generally better to keep the number of points as low as possible for the efficiency of the system. It is better to risk rendering a little too much than to spend the time culling objects at 20 different edges, for example.
+
+Room point editing
+~~~~~~~~~~~~~~~~~~
+You also have the option to manually edit the points used to define the convex hull of a room. These points are not present by default, you would typically create them by pressing the ``Generate Points`` button in the editor toolbar, when a room is selected. This will transfer the auto bound from the geometry (or manual ``Bound_`` mesh) into the inspector. Once there are points in the inspector, they will be used and override any other method. So if you wish to revert your manual editing, simply delete all the points.
```

Comment:
```suggestion
Room point editing
~~~~~~~~~~~~~~~~~~

You also have the option to manually edit the points used to define the convex hull of a room. These points are not present by default, you would typically create them by pressing the **Generate Points** button in the editor toolbar, when a room is selected. This will transfer the auto bound from the geometry (or manual ``Bound_`` mesh) into the inspector. Once there are points in the inspector, they will be used and override any other method. So if you wish to revert your manual editing, simply delete all the points.
```